### PR TITLE
Update RGFW and replace long switch in rcore_desktop_rgfw.c with a lookup table

### PR DESCRIPTION
**Pull request by ColleagueRiley**
I finally got around to writing a proper abstraction for keycodes for RGFW.  This PR includes that and many quality updates/fixes RGFW. 

JeffM commented on how `rcore_desktop_rgfw.c` uses a long switch for converting keycodes. That's not great for performance, this PR replaces it with a lookup table.

### CODE REVIEW

**Comment by ColleagueRiley** [comment]
Not sure why this claims there were 13 commits :sweat_smile:

File: src/platforms/rcore_desktop_rgfw.c [hunk 1]
```diff
@@ -114,6 +116,110 @@ extern CoreData CORE;                   // Global CORE state context
 
 static PlatformData platform = { NULL };   // Platform specific 
 
+static const u16 RGFWKeyToRayKey[] = {
```
Comment (raysan5) [inline src/platforms/rcore_desktop_rgfw.c:119]: In raylib I try to avoid typed data, please, could you use `unsigned short`?

**Comment by ColleagueRiley** [comment]
It should be good now :+1:

**Comment by raysan5** [comment]
@ColleagueRiley Nice! Thanks for the improvement!

**Pull request merged**